### DISCUSSION TITLE
[IOS Notification]HTTP/2‑based Apple Push Notification service (APNs)

### DISCUSSION THREAD

**Discussion by nobodyabcd** [invertase/react-native-firebase#4988] [General]
IOS announced that:

The HTTP/2‑based Apple Push Notification service (APNs) provider API lets you take advantage of great features, such as authentication with a JSON Web Token, improved error messaging, and per‑notification feedback. If you still send push notifications with the legacy binary protocol, make sure to upgrade to the APNs provider API as soon as possible. APNs will no longer support the legacy binary protocol after March 31, 2021.

Just curious, are we affected by this changes? 

https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/sending_notification_requests_to_apns

**Comment by mikehardy** [comment]
You should ask this on either firebase-ios-sdk or stackoverflow, or look for official upstream documentation. We won't have anything authoritative here, as this has to do with how the firebase FCM servers talk to the Apple APNs servers.
